### DISCUSSION TITLE
Implementing a Report API Button: Feedback and Suggestions on Design and Functionality

### DISCUSSION THREAD

**Discussion by FlavioAdamo** [Exifly/ApiVault#77] [Ideas]
👋 Hello everyone,

As part of our project, we are looking to implement a report API button in the API card. The purpose of this button is to allow users to report any issues or problems they encounter while using the API.

We would like to gather feedback and suggestions on how to implement this button in the best and simplest way possible, while also considering the design of the button itself.

Some questions to consider:

Where should the button be located in the API card?
What should be the text and/or icon on the button?
How should the button work? Should it open a modal, a new page, a issue on github? Should it prompt the user for additional information?

We appreciate your feedback and look forward to hearing your ideas on this topic.

**Comment by FlavioAdamo** [comment]
This is the solution implemented by @NirajD10, with the design given in the issue

I think that we can optimise the space in the card somehow, without having that extra bottom empty space.
Or maybe we can use that space con insert a like or dislike button.

Any ideas?

https://user-images.githubusercontent.com/66014487/235708065-2e43e566-f144-4827-9363-103b213933e4.mp4

**Reply by imggion** [reply]
Maybe the design can be a bit ugly with the report button only.

We can try to put in the tag section of an API so it will be on the same line.

Otherwise we can wait to implement an upvote system and add the report button with the upvote and downvote buttons as well.

**Reply by NirajD10** [reply]
I think I got idea. what about that the report button should appear when we hove on cards? it would be better than displaying all div. but although we have problem in mobile view.

**Reply by FlavioAdamo** [reply]
I love the hover idea. Maybe we can enable this function only on desktop mode for now, and then implement it on mobile when we will have the upvote/downvote system.

I think that we should not change the card height when hovering and the report button should appear in the bottom right corner as in the video.

What do you think?

**Reply by NirajD10** [reply]
yes, sounds good. for mobile view it likely possible we can implement cards display button whenever we are on display, like parallax scroll the report button will visible.

**Reply by imggion** [reply]
yeah i agree for mobile we should try to implement this solution, and i like the hover idea.

**Reply by NirajD10** [reply]
so lets wait upvote systems.. once two days is finished I'll try to see to implement it

**Reply by FlavioAdamo** [reply]
@NirajD10 
_yes, sounds good. for mobile view it likely possible we can implement cards display button whenever we are on display, like parallax scroll the report button will visible._

I love this idea

**Reply by NirajD10** [reply]
So should I start implement the feature?

**Reply by FlavioAdamo** [reply]
If you want and have time...yes! 
Happy coding!

**Reply by NirajD10** [reply]
I implemented desktop version. its good. but for mobile it's unlucky. I think it better display all report icon as each cards. 
or its there any ideas that we can implement on mobile version?

https://user-images.githubusercontent.com/66014487/236676501-b5866cb5-cf1f-4855-befc-42d171380b28.mp4

**Reply by NirajD10** [reply]
@FlavioAdamo @gdjohn4s ?

**Reply by NirajD10** [reply]
@FlavioAdamo @gdjohn4s ??
